modules/islandora_iiif/src/Plugin/views/style/IIIFManifest.php

```diff
@@ -312,71 +344,107 @@ protected function getTileSourceFromRow(ResultRow $row, $iiif_address, $iiif_bas
    * @return [string]
    *   The width and height of the image.
    */
-  protected function getCanvasDimensions(string $iiif_url, FieldItemInterface $image, string $mime_type) {
+  protected function getCanvasDimensions(string $iiif_url, Media $media, FieldItemInterface $image, string $mime_type) {
+
+    // If the media has field_height and field_width, return those values.
+    if ($media->hasField('field_height')
+      && !$media->get('field_height')->isEmpty()
+      && $media->get('field_height')->value > 0
+      && $media->hasField('field_width')
+      && !$media->get('field_width')->isEmpty()
+      && $media->get('field_width')->value > 0) {
+      return [intval($media->get('field_width')->value),
+        intval($media->get('field_height')->value),
+      ];
+    }
```

Comment:
Not sure this makes sense to do _before_ attempting to acquire the values from `$image`, is there's a decent chance that `$image->{width,height}` are closer to the source of truth: https://git.drupalcode.org/project/drupal/-/blob/10.2.x/core/modules/image/src/Plugin/Field/FieldType/ImageItem.php?ref_type=heads#L156-162

IIRC, part of the media ingest process can indeed copy these properties to fields on the media?, (i.e.: https://github.com/Islandora-Devops/islandora-starter-site/blob/52a1f3aa1aa52e7652986f53ad3f174bf6423ac1/config/sync/media.type.image.yml#L19-L20)

Also, this is already effectively accounted for below?: https://github.com/Islandora/islandora/pull/983/files#diff-edc92719b6b111e26d43c711272a79678edfb7defa5705a23f0dc70292e1898dR377-R386

If there's a desire to reweight the particular mechanisms, it might make sense to things out to events, or some kind of weighted plugin collection mechanism?

Reply:
We're running into a situation with tiff paged objects where the tiffs fall through all these conditions and end up having getimgsize run on them over and over.  Even though they already had their width/height calculated and stored on the media.

I'm more or less just swooping in from fixing a running site and am putting this up for @alxp . Wasn't trying to introduce any architecture changes per se.  Though I do feel we should short circuit out as soon as possible to avoid getimgsize or, even worse, talking to cantaloupe.

FWIW, just doing this took a 432 page book's manifest's load time from a minute and a half to 13 seconds.

Reply:
It then sounds like this code wasn't present in the environment in the first place, as [acquisition from the parent entity](https://github.com/Islandora/islandora/pull/983/files#diff-edc92719b6b111e26d43c711272a79678edfb7defa5705a23f0dc70292e1898dR377-R386) already precedes the [`getimagesize()` processing](https://github.com/Islandora/islandora/pull/983/files#diff-edc92719b6b111e26d43c711272a79678edfb7defa5705a23f0dc70292e1898dR388-R401) calls?